### PR TITLE
JBIDE-20071 Allow deploying exploded jars to war/WEB-INF/lib

### PR DESCRIPTION
**Pull request by VsevolodGolovanov**
Webfragments and EJBs in WARs deploy exploded.

### CODE REVIEW

File: as/plugins/org.jboss.ide.eclipse.as.core/jbosscore/org/jboss/ide/eclipse/as/core/publishers/PublishUtil.java [hunk 1]
```diff
@@ -64,13 +64,6 @@ public static boolean deployPackaged(IModule[] moduleTree) {
 			return true;
 		} else if( moduleTypeId.equals(IWTPConstants.FACET_APP_CLIENT)) { 
 			return true;
-		} else if( moduleTypeId.equals(IWTPConstants.FACET_WEB_FRAGMENT)) {
```
Comment (maxandersen) [inline as/plugins/org.jboss.ide.eclipse.as.core/jbosscore/org/jboss/ide/eclipse/as/core/publishers/PublishUtil.java]: this logic should depend on the version of the server should it not ? I think there was a reason for why this was done?

**Comment by VsevolodGolovanov** [comment]
@maxandersen, I don't know, should it? I'm not sure about the extent to which JBoss Tools go to support the old versions. Both Webfragments and EJBs in WARs seem to be rarely used things. And the reason was fixed 3 years ago: https://issues.jboss.org/browse/AS7-4704. If someone is affected by the change, then he can just settle on a known working JBoss Tools version.

If that's unsatisfactory, I'll try to implement the version check. Ideally the change should only apply to AS 7.1.3+, AS 7.2.0+ and WF.

**Comment by maxandersen** [comment]
you'll notice we have server adapters from JBoss 3 - 6, Wildfly 7-10 and EAP 4 to 6 - we don't want to force users to have to install custom IDE's just to maintain some old version.

So yeah, version of server matters.

**Comment by maxandersen** [comment]
note: Rob is on PTO right now so might be a bit before he can follow up to verify/reject my concerns.

**Comment by VsevolodGolovanov** [comment]
@maxandersen, added server version checking.
Tested publishing to AS 5.1.0, AS 6.1.0, AS 7.1.1 - EJBs in WARs and WebFragments deploy packaged.
EAP 6.1.0, WF 8.2.0, WF 9.0.0 - EJBs in WARs and WebFragments deploy exploded.

Couldn't test EAP 6.0.0 (should be packaged) and 6.0.1 (should be exploded), because they're not available for download.

**Comment by robstryker** [comment]
This is a really solid patch.  I will make one change though:  changing the static method's signature really isn't allowed (API), but we can add a new signature with the server, and have the old one call the new one with a null passed in.

**Comment by robstryker** [comment]
I've pushed with some small additions. Great patch.

**Pull request closed without merging**